### PR TITLE
Add Configuration Tree building support.

### PR DESCRIPTION
**Pull request by Djeg**
So this is my version of TreeBuilding implementation. Let's take an exemple :

``` php
<?php

namespace App;

use Knp\RadBundle\AppBundle\Bundle;
use Symfony\Component\Config\Definition\Builder\NodeParentInterface;
use Symfony\Component\DependencyInjection\ContainerBuilder;

class App extends Bundle
{
    public function buildConfiguration(NodeParentInterface $rootNode)
    {
        $rootNode
            ->children()
                ->scalarNode('Foo')
                    ->defaultValue("bar")
                    ->info('This is a fuck*** bar !!')
                ->end()
            ->end()
        ;
    }

    public function buildContainer(array $config, ContainerBuilder $container)
    {
        if (isset($config['foo'])) {
            $container->addParameter('some.foo', $config['foo']);
        }
    }
}
```

This is a default Bundle called "App" with an implementation of my version of TreeBuilding simplification. Of course, we can erased this two method from this bundle. This PR don't break the compatibility. Just add the TreeBuilder support.

### CODE REVIEW

File: AppBundle/ConfigurableBundleInterface.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Knp\RadBundle\AppBundle;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\Definition\Builder\NodeParentInterface;
+
+interface ConfigurableBundleInterface
```
Comment (docteurklein) [inline AppBundle/ConfigurableBundleInterface.php]: why `Bundle` in the name ? it has nothing to do with bundle. This interface could be used anywhere else.

File: AppBundle/Bundle.php [hunk 2]
```diff
@@ -29,6 +30,16 @@ public function getContainerExtension()
             return $extension;
         }
 
-        return $this->extension = new ContainerExtension($this->getPath());
+        return $this->extension = new ContainerExtension($this);
+    }
+
+    public function buildConfiguration(NodeParentInterface $rootNode)
+    {
+        return;
```
Comment (docteurklein) [inline AppBundle/Bundle.php]: return are useless here.

File: AppBundle/ConfigurableBundleInterface.php (hunk 1 above)
Comment (Djeg) [inline AppBundle/ConfigurableBundleInterface.php]: Yes exact ;) !

**Comment by docteurklein** [comment]
Otherwise, it seems good! Please add some tests if possible.

File: AppBundle/Configuration.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	private $bundle;
+
+	public function getConfigTreeBuilder()
+	{
+		$treeBuilder = new TreeBuilder();
+
+        $rootNode = $treeBuilder->root('app');
+        $this->bundle->buildConfiguration($rootNode);
+
+        return $treeBuilder;
```
Comment (gquemener) [inline AppBundle/Configuration.php:32]: Hi David, please fix the indentation here.

File: AppBundle/Configuration.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	}
+
+	public function __construct(Bundle $bundle)
```
Comment (gquemener) [inline AppBundle/Configuration.php]: Please put the constructor as the first method also

File: AppBundle/Configuration.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		$this->bundle = $bundle;
```
Comment (gquemener) [inline AppBundle/Configuration.php]: And fix the indentation here (4 spaces)

File: AppBundle/Bundle.php [hunk 6]
```diff
@@ -29,6 +30,16 @@ public function getContainerExtension()
...
+    }
+
+    public function buildContainer(array $config, ContainerBuilder $container)
+    {
+        return;
```
Comment (gquemener) [inline AppBundle/Bundle.php]: Here also

File: AppBundle/ConfigurableBundleInterface.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Knp\RadBundle\AppBundle;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\Definition\Builder\NodeParentInterface;
+
+interface ConfigurableBundleInterface 
+{
+	/**
```
Comment (gquemener) [inline AppBundle/ConfigurableBundleInterface.php]: Indentation is also broken here

**Comment by gquemener** [comment]
Ok for me after fixing cs related issues (see my comments), and adding/updating tests.

**Comment by Djeg** [comment]
Does it seems good ? Composer seems to crash for the SF 2.3 version on travis :(

**Comment by lsmith77** [comment]
see #91

**Comment by gquemener** [comment]
Seems good to me, but no time to test it right now. @docteurklein @Herzult

**Comment by docteurklein** [comment]
I'm on it. Meanwhile, a 2.3 branch has been created to handle symfony 2.3 deps.

**Comment by docteurklein** [comment]
oops merged while it needed to refactor specs to handle phpspec/phpspec@develop

**Comment by docteurklein** [comment]
@davidjegat can you PR with the new phpspecs please?

**Pull request merged**